app/components/HamburgerMenu.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+import {
+  Flex,
+  HStack,
+  Spacer,
+  Text,
+  Button,
+  IconButton,
+} from "@chakra-ui/react";
+import { useState } from "react";
+import { IoClose } from "react-icons/io5";
+import { IoMenu } from "react-icons/io5";
+import { usePathname } from "next/navigation";
+
+import Link from "next/link";
+
+const HamburgerMenu = () => {
+  const pathname = usePathname();
+  const [displayMenu, setDisplayMenu] = useState("none");
```

Comment:
Om du vill ha typat displayMenu här, så kan du skriva
```suggestion
  const [displayMenu, setDisplayMenu] = useState<CSSProperties['display']>("none");
```
eller
```suggestion
  const [displayMenu, setDisplayMenu] = useState<'none'|'flex'>("none");
```